src/test/java/org/apache/sysds/test/component/matrix/EigenDecompTest.java

```diff
@@ -0,0 +1,112 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.component.matrix;
+
+import org.apache.sysds.runtime.matrix.data.LibCommonsMath;
+import org.apache.sysds.runtime.matrix.data.MatrixBlock;
+import org.apache.sysds.runtime.util.DataConverter;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+import org.junit.Assert;
+
+public class EigenDecompTest {
+
+	@Test public void testLanczosSimple() {
+		double tol = 1e-4;
+
+		MatrixBlock in = new MatrixBlock(4, 4, false);
+		double[] a = { 4, 1, -2,  2,
+					   1, 2,  0,  1,
+		              -2, 0,  3, -2,
+				       2, 1, -2, -1};
+		in.init(a, 4, 4);
+		MatrixBlock[] m1 = LibCommonsMath.multiReturnOperations(in, "eigen");
+		MatrixBlock[] m2 = LibCommonsMath.multiReturnOperations(in, "eigen_lanczos");
+
+		TestUtils.compareMatrices(m1[0], m2[0], tol, "Result of eigenvalues of new eigen_lanczos function wrong");
+		testEvecValues(m1[1], m2[1], tol);
+	}
+
+	@Test public void testLanczosRandom() {
+		double tol = 1e-4;
+
+		MatrixBlock in = TestUtils.generateTestMatrixBlockSym(10, 10, 0.0, 1.0, 1.0, 1);
+		// MatrixBlock in = TestUtils.generateTestMatrixBlockSym(100, 100, 0.0, 1.0, 1.0, 1); // fails
+		long t1 = System.nanoTime();
+		MatrixBlock[] m1 = LibCommonsMath.multiReturnOperations(in, "eigen");
+		long t2 = System.nanoTime();
+		MatrixBlock[] m2 = LibCommonsMath.multiReturnOperations(in, "eigen_lanczos");
+		long t3 = System.nanoTime();
+
+		System.out.println(
+			"time eigen: " + (t2 - t1) + " time Lanczos: " + (t3 - t2) + " Lanczos speedup: " + ((double) (t2 - t1) / (t3 - t2)));
+		TestUtils.compareMatrices(m1[0], m2[0], tol, "Result of eigenvalues of new eigen_lanczos function wrong");
+		testEvecValues(m1[1], m2[1], tol);
+	}
+
+	@Test public void testQREigenSimple() {
+		double tol = 1e-4;
+
+		MatrixBlock in = new MatrixBlock(4, 4, false);
+		double[] a = { 52, 30, 49, 28,
+				       30, 50,  8, 44,
+				       49,  8, 46, 16,
+					   28, 44, 16, 22};
+		in.init(a, 4, 4);
+
+		MatrixBlock[] m1 = LibCommonsMath.multiReturnOperations(in, "eigen");
+		MatrixBlock[] m2 = LibCommonsMath.multiReturnOperations(in, "eigen_qr");
+
+		TestUtils.compareMatrices(m1[0], m2[0], tol, "Result of eigenvalues of new eigen_qr function wrong");
+		testEvecValues(m1[1], m2[1], tol);
+	}
+
+	@Test public void testQREigenRandom() {
+		double tol = 1e-4;
+
+		MatrixBlock in = TestUtils.generateTestMatrixBlockSym(10, 10, 0.0, 1.0, 1.0, 1);
+		// MatrixBlock in = TestUtils.generateTestMatrixBlock(5, 5, 0.0, 1.0, 1.0, 5); // fails evals correct evecs wrong (evec corresponding to largest eval correct)
+		// MatrixBlock in = TestUtils.generateTestMatrixBlock(10, 10, 0.0, 1.0, 1.0, 2); // fails complex EVs
+		// MatrixBlock in = TestUtils.generateTestMatrixBlockSym(50, 50, 0.0, 1.0, 1.0, 1); // fails not converged
```

Comment:
instead of leaving this outcommented, add more tests, each calling another method that execute the test with the given argument MatrixBlock "in"